src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -492,12 +492,19 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
-        return sprintf('alter table %s add %s %s%s(%s)',
+        $columns = collect($command->columns)->map(function ($column) {
+            // Check if the column contains a functional expression
+            // and return as-is, otherwise wrap
+            return preg_match('/\(.+\)/', $column) ? $column : $this->wrap($column);
```

Comment:
https://github.com/laravel/framework/pull/54025#discussion_r1898688984 can be applied here, then:
```suggestion
            return self::isFunctionalExpression($column) ? $column : $this->wrap($column);
```